### PR TITLE
Add functionality for creating loops with an intro

### PR DESCRIPTION
**Pull request by Wrymouth**

### CODE REVIEW

File: mamar-web/src/app/store/variation.ts [hunk 1]
```diff
@@ -73,6 +81,32 @@ export function variationReducer(variation: Variation, action: VariationAction):
...
+        const startLoopSeg: Segment = {
+            id: action.id,
+            type: "StartLoop",
+            label_index: action.labelIndex,
+        }
+        return {
+            ...variation,
+            segments: [
+                ...variation.segments,
+                startLoopSeg,
```
Comment (nanaian) [inline mamar-web/src/app/store/variation.ts:107]: Should probably swap these two lines. StartLoop needs to go at the start. Also should add EndLoop, as having a StartLoop without an EndLoop after it is an error

Comment (Wrymouth) [reply mamar-web/src/app/store/variation.ts:107]: StartLoop shouldn't go at the start of the entire song, right? That seems like it'd go against the point of this PR

Comment (nanaian) [reply mamar-web/src/app/store/variation.ts:107]: It should be configurable yeah. But certainly not at the end, there should be a segment between it

Perhaps add
StartLoop
empty segment
EndLoop

Comment (Wrymouth) [reply mamar-web/src/app/store/variation.ts:107]: Sounds good, provided you can then import a midi to be used by that empty segment

**Review by bates64 — APPROVED** [review]

**Pull request merged**